perf_test/blas/blas3/KokkosBlas3_gemm_perf_test.hpp

```diff
@@ -0,0 +1,1015 @@
+                                                        gemm_args_.beta,
+                                                        svC_row);
+        });
+  }
+};
+
+/**
+ * 2. case a)
+ * parallel_for(teamPolicy): TeamVectorGemm
+ *
+ */
+template <class TransAType, class TransBType, class BlockingType,
+          class device_type>
+void __do_gemm_parallel_experiment2(options_t options, gemm_args_t gemm_args) {
+  using execution_space = typename device_type::execution_space;
+  using policy_type     = Kokkos::TeamPolicy<TeamVectorTag, execution_space>;
+  using member_type     = typename policy_type::member_type;
+  using functor_type =
+      parallel_batched_gemm_experiment2_3_4<TransAType, TransBType,
+                                            BlockingType, member_type>;
+
+  uint32_t warm_up_n = options.warm_up_n;
+  uint32_t n         = options.n;
+  auto league_size   = options.start.c.k;
+  Kokkos::Timer timer;
+  STATUS;
+
+  functor_type experiment2_functor(gemm_args);
+
+  auto team_size  = gemm_args.bp.team_size;
+  auto vector_len = gemm_args.bp.vector_len;
+
+  for (uint32_t i = 0; i < warm_up_n; ++i) {
+    Kokkos::parallel_for("parallelBatchedUntimedExperiment2Gemm",
+                         policy_type(league_size, team_size, vector_len),
+                         experiment2_functor);
+  }
+  Kokkos::fence();
+
+  timer.reset();
+
+  for (uint32_t i = 0; i < n; ++i) {
+    Kokkos::parallel_for("parallelBatchedTimedExperiment2Gemm",
+                         policy_type(league_size, team_size, vector_len),
+                         experiment2_functor);
+  }
+  Kokkos::fence();
+
+  __gemm_output_csv_row(options, gemm_args, timer.seconds(), "experiment2");
+  return;
+}
+
+/**
+ * 3. case b)
+ *    parallel_for(teamPolicy):
+ *      parallel_for(TeamThreadRange):
+ *         VectorGemm
+ *
+ * VectorGemm has not been implemented!
+ * I think this experiment can be removed. TeamGemm calls TeamThreadRange
+ * internally! TeamVectorGemm calls both TeamThreadRange and ThreadVectorRange
+ * internnally!
+ */
+template <class TransAType, class TransBType, class BlockingType,
+          class device_type>
+void __do_gemm_parallel_experiment3(options_t options, gemm_args_t gemm_args) {
+  using execution_space = typename device_type::execution_space;
+  // using layout_tag = std::conditional<std::is_same<default_layout,
+  // Kokkos::LayoutLeft>::value, LayoutLeftTag, LayoutRightTag>::type;
+  using policy_type = Kokkos::TeamPolicy<LayoutLeftTag, execution_space>;
+  using member_type = typename policy_type::member_type;
+  using functor_type =
+      parallel_batched_gemm_experiment2_3_4<TransAType, TransBType,
+                                            BlockingType, member_type>;
+
+  uint32_t warm_up_n = options.warm_up_n;
+  uint32_t n         = options.n;
+  auto league_size   = options.start.c.k;
+  Kokkos::Timer timer;
+  STATUS;
+
+  functor_type experiment4_functor(gemm_args);
```

Comment:
I think this functor name is mislabeled for experiment4 and should be experiment3? (The LayoutLeftTag is listed under experiment 3 of your operator overloads for `parallel_batched_gemm_experiment2_3_4`)
If mislabeled, the functor objects passed to parallel_for's below will need to be renamed also